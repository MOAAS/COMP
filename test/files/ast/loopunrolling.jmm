import static io.println(int);

class LoopUnrolling {
	public static void main(String[] args) {
	    int[] array;
	    int i;

	    array = new int[100];
	    i = 0;
	    while (i < 100) {
	        array[i] = i;
	        i = i + 1;
	    }

        i = 0;
	    while (i < 100) {
	        io.println(array[i]);
	        i = i + 1;
	    }

        i = 3;
	    while (i < 20) {
	        io.println(array[i]);
	        i = i + 2;
	    }

	   	/*
            while (i < 100) {
                io.println(array[i]);
                i = i + 1;
                io.println(array[i]);
                i = i + 1;
                io.println(array[i]);
                i = i + 1;
                io.println(array[i]);
                i = i + 1;
                io.println(array[i]);
                i = i + 1;
            }
	    */


	}
}