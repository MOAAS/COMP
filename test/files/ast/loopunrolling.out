Program
 ImportDeclaration
  ImportLifetime
  ImportName
   io
   println
  ImportArgs
   int
  ImportType
 ClassDeclaration
  ClassInfo
   LoopUnrolling
  MainDeclaration
   ArgName
    args
   MethodContent
    VarDeclaration
     int[]
     array
    VarDeclaration
     int
     i
    Assignment
     array
     ArrayInit
      100
    Assignment
     i
     0
    While
     LessThan
      i
      100
     CodeBlock
      Assignment
       Access
        array
        i
       i
      Assignment
       i
       Add
        i
        1
      Assignment
       Access
        array
        i
       i
      Assignment
       i
       Add
        i
        1
      Assignment
       Access
        array
        i
       i
      Assignment
       i
       Add
        i
        1
      Assignment
       Access
        array
        i
       i
      Assignment
       i
       Add
        i
        1
      Assignment
       Access
        array
        i
       i
      Assignment
       i
       Add
        i
        1
    Assignment
     i
     0
    While
     LessThan
      i
      100
     CodeBlock
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        1
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        1
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        1
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        1
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        1
    Assignment
     i
     3
    While
     LessThan
      i
      20
     CodeBlock
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        2
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        2
      Dot
       io
       DotArgs
        println
        Access
         array
         i
      Assignment
       i
       Add
        i
        2
